chain/events/filter/event.go

```diff
@@ -27,22 +27,32 @@ func isIndexedValue(b uint8) bool {
 	return b&(types.EventFlagIndexedKey|types.EventFlagIndexedValue) > 0
 }
 
-type EventFilter struct {
-	id         types.FilterID
-	minHeight  abi.ChainEpoch // minimum epoch to apply filter or -1 if no minimum
-	maxHeight  abi.ChainEpoch // maximum epoch to apply filter or -1 if no maximum
-	tipsetCid  cid.Cid
-	addresses  []address.Address   // list of f4 actor addresses that are extpected to emit the event
-	keys       map[string][][]byte // map of key names to a list of alternate values that may match
-	maxResults int                 // maximum number of results to collect, 0 is unlimited
+type AddressResolver func(context.Context, abi.ActorID, *types.TipSet) (address.Address, bool)
+
+type EventFilter interface {
+	Filter
+
+	TakeCollectedEvents(context.Context) []*CollectedEvent
+	CollectEvents(context.Context, *TipSetEvents, bool, AddressResolver) error
+}
+
+type eventFilter struct {
+	id        types.FilterID
+	minHeight abi.ChainEpoch // minimum epoch to apply filter or -1 if no minimum
+	maxHeight abi.ChainEpoch // maximum epoch to apply filter or -1 if no maximum
+	tipsetCid cid.Cid
+	addresses []address.Address // list of f4 actor addresses that are extpected to emit the event
```

Comment:
Why is this only applicable to f4 actors..?

Reply:
outdated comment from eth-specific api, removing